File: Android_Zone_Lib/src/com/zone/lib/utils/activity_fragment_ui/ToastUtils.java
package com.zone.lib.utils.activity_fragment_ui;

import android.content.Context;
import android.os.Handler;
import android.os.Looper;
import android.widget.Toast;

import com.zone.lib.Configure;

/**
 * Toast统一管理类
 * 
 */
public class ToastUtils {
	private static boolean showToast = true;

	public static void showToast(final int msgId) {
		if (Looper.getMainLooper() != Looper.myLooper()) {
			Handler handler = new Handler(Looper.getMainLooper());
			handler.post(new Runnable() {
				@Override
				public void run() {
					showShort(Configure.getApplicationContext(), msgId);
				}
			});
		} else {
			showShort(Configure.getApplicationContext(), msgId);
		}
	}

	/**
	 * 短时间显示Toast
	 * 
	 * @param context
	 * @param message
	 */
	public static void showShort(Context context, CharSequence message)
	{
		if (showToast)
			Toast.makeText(context, message, Toast.LENGTH_SHORT).show();
	}

	/**
	 * 短时间显示Toast
	 * 
	 * @param context
	 * @param message
	 */
	public static void showShort(Context context, int message)
	{
		if (showToast)
			Toast.makeText(context, message, Toast.LENGTH_SHORT).show();
	}

	/**
	 * 长时间显示Toast
	 * 
	 * @param context
	 * @param message
	 */
	public static void showLong(Context context, CharSequence message)
	{
		if (showToast)
			Toast.makeText(context, message, Toast.LENGTH_LONG).show();
	}

	/**
	 * 长时间显示Toast
	 * 
	 * @param context
	 * @param message
	 */
	public static void showLong(Context context, int message)
	{
		if (showToast)
			Toast.makeText(context, message, Toast.LENGTH_LONG).show();
	}

	/**
	 * 自定义显示Toast时间
	 * 
	 * @param context
	 * @param message
	 * @param duration
	 */
	public static void show(Context context, CharSequence message, int duration)
	{
		if (showToast)
			Toast.makeText(context, message, duration).show();
	}

	/**
	 * 自定义显示Toast时间
	 * 
	 * @param context
	 * @param message
	 * @param duration
	 */
	public static void show(Context context, int message, int duration)
	{
		if (showToast)
			Toast.makeText(context, message, duration).show();
	}
	public static void openToast(){
		showToast=true;
	}
	public static void closeToast(){
		showToast=false;
	}
}

File: Android_Zone_Lib/src/com/zone/lib/utils/activity_fragment_ui/ToastUtils.kt
package com.zone.lib.utils.activity_fragment_ui

import android.content.Context
import android.widget.Toast

/**
 * Toast统一管理类
 *
 * //todo zone  统一处理异常  需要有个异常管理类。 例如 json转换错误应该显示什么呢？当然数据应该弄到反馈里不然也修复不了
 *      统一过滤？类似okhttp的拦截器？
 *      可以子线程中使用
 *      https://www.jianshu.com/p/d6c9a485c061   某个view为基点的 gravity
 *      自定义view 可以 有动画么？
 */
object ToastUtils {
    private var showToast = true

    /**
     * 短时间显示Toast
     *
     * @param context
     * @param message
     */
    fun showShort(context: Context, message: CharSequence) {
        if (showToast)
            Toast.makeText(context, message, Toast.LENGTH_SHORT).show()
    }

    /**
     * 短时间显示Toast
     *
     * @param context
     * @param message
     */
    fun showShort(context: Context, message: Int) {
        if (showToast)
            Toast.makeText(context, message, Toast.LENGTH_SHORT).show()
    }

    /**
     * 长时间显示Toast
     *
     * @param context
     * @param message
     */
    fun showLong(context: Context, message: CharSequence) {
        if (showToast)
            Toast.makeText(context, message, Toast.LENGTH_LONG).show()
    }

    /**
     * 长时间显示Toast
     *
     * @param context
     * @param message
     */
    fun showLong(context: Context, message: Int) {
        if (showToast)
            Toast.makeText(context, message, Toast.LENGTH_LONG).show()
    }

    /**
     * 自定义显示Toast时间
     *
     * @param context
     * @param message
     * @param duration
     */
    fun show(context: Context, message: CharSequence, duration: Int) {
        if (showToast)
            Toast.makeText(context, message, duration).show()
    }

    /**
     * 自定义显示Toast时间
     *
     * @param context
     * @param message
     * @param duration
     */
    fun show(context: Context, message: Int, duration: Int) {
        if (showToast)
            Toast.makeText(context, message, duration).show()
    }



    fun openToast() {
        showToast = true
    }

    fun closeToast() {
        showToast = false
    }
}
